src/neo-vm/Types/Buffer.cs

```diff
@@ -40,22 +40,6 @@ public override StackItem ConvertTo(StackItemType type)
             }
         }
 
-        public override bool Equals(object obj)
-        {
-            return this == obj;
-        }
```

Comment:
Instead of deleting should we do reference + sequence equality checking (=the same as `ByteString`)? 

Now it is only reference equality checking via `StackItem`.

Reply:
@shargon What do you think?

Reply:
We are temporarily unable to reach a consensus on this issue, and the function here has not been modified, it is still the same as the original. So I will merge this PR first. If there is a problem, you can create a new issue separately.

Reply:
is @shargon afk or did he share his opinion elsewhere?

Reply:
Sorry i miss this conversation. 

> reference + sequence equality checking 

If it's the same reference, the sequence will be the same, but reference it's good for me

Reply:
@shargon The point I'm trying to make is that the immutable version (`ByteString`) does reference + sequence checking, but the mutable version `Buffer` does not. Which I think should be the same, and you previous

Reply:
It could be easier, incompatible with c# but easy for developers. Use `==` and check if the buffer it's the same. But also... do you want to be able to compare ByteString with Buffer?

Reply:
as described above I can imagine a case where you'd like to compare 2 `Buffers` on sequence without having to convert them to `ByteString` first. It doesn't sound logical to me that that sequence comparison only works in the immutable variant.